include/seqan3/io/sequence_file/output.hpp

```diff
@@ -339,37 +339,56 @@ class sequence_file_output
+    sequence_file_output(filesystem::path filename,
+                         selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{}) :
+        primary_stream{new std::ofstream{filename, std::ios_base::out | std::ios::binary}, stream_deleter_default}
     {
-        // open stream
-        stream.open(_file_name, std::ios_base::out | std::ios::binary);
-        if (!stream.is_open())
+        if (!primary_stream->good())
             throw file_open_error{"Could not open file for writing."};
 
+        // possibly add intermediate compression stream
+        secondary_stream = detail::make_secondary_ostream(*primary_stream, filename);
+
         // initialise format handler or throw if format is not found
-        detail::set_format(format, _file_name);
+        detail::set_format(format, filename);
     }
 
     /*!\brief Construct from an existing stream and with specified format.
      * \tparam file_format   The format of the file in the stream, must satisfy seqan3::sequence_file_output_format_concept.
-     * \param[in] _stream    The stream to operate on (this must be std::move'd in!).
+     * \param[in,out] stream The stream to write to, must be derived of std::basic_ostream<stream_char_t>.
      * \param[in] format_tag The file format tag.
      * \param[in] fields_tag A seqan3::fields tag. [optional]
      */
-    template <sequence_file_output_format_concept file_format>
-    sequence_file_output(stream_type             && _stream,
-                      file_format        const & SEQAN3_DOXYGEN_ONLY(format_tag),
-                      selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{}) :
-        stream{std::move(_stream)}, format{file_format{}}
+    template <ostream_concept2 stream_t,
+              sequence_file_output_format_concept file_format>
+    sequence_file_output(stream_t                 & stream,
+                         file_format        const & SEQAN3_DOXYGEN_ONLY(format_tag),
+                         selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{}) :
+        primary_stream{&stream, stream_deleter_noop},
+        secondary_stream{&stream, stream_deleter_noop},
+        format{file_format{}}
+    {
+        static_assert(meta::in<valid_formats, file_format>::value,
+                      "You selected a format that is not in the valid_formats of this file.");
+    }
+
+    //!\overload
+    template <ostream_concept2 stream_t,
+              sequence_file_output_format_concept file_format>
+    sequence_file_output(stream_t                && stream,
+                         file_format        const & SEQAN3_DOXYGEN_ONLY(format_tag),
+                         selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{}) :
+        primary_stream{new stream_t{std::move(stream)}, stream_deleter_default},
+        secondary_stream{&*primary_stream, stream_deleter_noop},
+        format{file_format{}}
```

Comment:
I must have missed the point where you call make_secondary_ostream. Or is it not required here? Please enlight me.

Reply:
It's not possible, because how would you know which compression the user wants? But I have now added documentation sections to every constructor that say whether it does transparent (de)compression or not.